src/main/java/ViewModel/playlistViewModel.java

```diff
@@ -1,4 +1,37 @@
 package ViewModel;
+import Controller.newPlaylistController;
+import Entities.RegularUser;
+import Entities.Song;
+import InputData.newPlaylistInputData;
+import OutputData.newPlaylistOutputData;
+import UseCase.createPlaylist;
 
+
+import Presenter.playlistPresenter;
 public class playlistViewModel {
+    private final newPlaylistController newPlaylistController;
+    private  playlistPresenter playlistPresenter;
+
+    public playlistViewModel(int id, RegularUser user, String playlistName, Song firstSong){
+        // get necessary information from View to construct a new playlist
+        newPlaylistInputData inputData = new newPlaylistInputData(id,playlistName,firstSong,user);
+        //Blank output data object to extract success message from
+        newPlaylistOutputData outputData = new newPlaylistOutputData("");
```

Comment:
I don't think you should construct the OutputData in the presenter. I could be wrong, however.

Reply:
If that's the case, Presenter does not need to take OutputData as a parameter.